sklearn/ensemble/voting_classifier.py

```diff
@@ -0,0 +1,206 @@
+    >>> np.random.seed(123)
+    >>> clf1 = LogisticRegression()
+    >>> clf2 = RandomForestClassifier()
+    >>> clf3 = GaussianNB()
+    >>> X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
+    >>> y = np.array([1, 1, 1, 2, 2, 2])
+    >>> eclf1 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='hard')
+    >>> eclf1 = eclf1.fit(X, y)
+    >>> print(eclf1.predict(X))
+    [1 1 1 2 2 2]
+    >>> eclf2 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='soft')
+    >>> eclf2 = eclf2.fit(X, y)
+    >>> print(eclf2.predict(X))
+    [1 1 1 2 2 2]
+    >>> eclf3 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='soft', weights=[2,1,1])
+    >>> eclf3 = eclf3.fit(X, y)
+    >>> print(eclf3.predict(X))
+    [1 1 1 2 2 2]
+    >>>
+      
+    """
+    def __init__(self, estimators, voting='hard', weights=None):
+        self.estimators = estimators
+        self.voting = voting
+        self.weights = weights
+        self.le = LabelEncoder()
+
+    
+    def fit(self, X, y):
+        """ 
+        Fits the estimators.
+        
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+            
+        y : array-like, shape = [n_samples]
+            Target values.
+      
+        Returns
+        -------
+        self : object
+
+        """
+        
+        if self.voting not in ('soft', 'hard'):
+            raise ValueError("voting must be 'soft' or 'hard'; got (voting=%r)"
+                             % voting)
+                                                  
+        if self.weights and len(self.weights) != len(self.estimators):
+            raise ValueError('Number of classifiers and weights must be equal'\
+                         '; got %d weights, %d estimators' 
+                         % (len(self.weights), len(self.estimators)))      
+        
+        for clf in self.estimators:
+            clf.fit(X, y)
+        
+        self.le.fit(y)
+        self.classes_ = np.unique(y)
+                
+        return self
+            
+    
+    def predict(self, X):
+        """
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        
+        Returns
+        ----------
+        maj : array-like, shape = [n_cl_lab]
+            Predicted class labels by majority rule.
+        
+        """
+        if self.voting == 'soft':
+            avg = self.predict_proba(X)
+            maj = self.le.inverse_transform(np.argmax(avg, axis=1))
+        
+        else: # 'hard' voting
+            cl_lab = self._predict(X)
+            
+            # arange class labels as [n_cl_lab, n_classifier_predictions]
```

Comment:
I couldn't find a meaningful description for it. E.g., if 3 classifiers are used in the VotingClassifier, this would look like
    
array([[0, 0, 0],
           [0, 0, 0],
          ...,
           [2, 2, 2]])

where the 1st column are the class labels predicted by the 1st classifier, the 2nd column the class labels predicted by the 2nd classifier etc.

Maybe something like `[n_classifier_results, n_classifier]` is more meaningful?